### PR TITLE
Remove redesigned lists feature flag and govuk_admin_template gem

### PR DESCRIPTION
**Pull request by davidgisbey**
## Description

Recently, we've done some work to rebuild the curated lists pages. Now that has been completed we can remove the legacy implementation.

This pr removes all list related legacy code, the `Redesigned list` feature flag and the govk_admin_template gem.

## Trello card 

https://trello.com/c/dLRsBjVj/427-remove-govukadmintemplate-from-collections-publisher

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: app/controllers/lists_controller.rb [hunk 1]
```diff
@@ -1,13 +1,7 @@
 class ListsController < ApplicationController
-  layout :get_layout
   before_action :find_tag
```
Comment (ollietreend) [inline app/controllers/lists_controller.rb:2]: I can see a lot of repetition in this controller:

```ruby
@list = @tag.lists.find(params[:id])
```

I'm wondering if this controller would also benefit from a `find_list` method, the same as `ListItemsController` has.

File: app/views/lists/start_curating_lists.html.erb [hunk 2]
```diff
@@ -1,14 +0,0 @@
-<h2>Start curating links for '<%= @tag.title %>'</h2>
-
-<div class='row'>
-  <div class='col-md-6'>
-    <div class="lead">
-      Items tagged with '<%= @tag.title %>' are
-      <%= link_to 'currently displayed in alphabetical order',
-      @tag.web_url, target: '_blank' %> ("A-to-Z"). By creating the first list
-      you can start organizing the items.
```
Comment (ollietreend) [inline app/views/lists/start_curating_lists.html.erb:9]: Spelling mistake! ~organizing~ organising 😱

**Review by ollietreend — COMMENTED** [review]
Nice work tidying this all up ✨🧹

Just a couple of comments from me.

File: app/controllers/lists_controller.rb (hunk 1 above)
Comment (davidgisbey) [reply app/controllers/lists_controller.rb:2]: will do 👍

Comment (ollietreend) [reply app/controllers/lists_controller.rb:2]: @davidgisbey we should be able to DRY out these actions now:

- show
- edit
- confirm_destroy
- update
- edit_list_items
- update_list_items
- manage_list_item_ordering
- update_list_item_ordering

Comment (davidgisbey) [reply app/controllers/lists_controller.rb:2]: sorry got distracted half way through fixing this 🤦 Have pushed it up!

**Review by ollietreend — APPROVED** [review]
Looks great 👍🏻

**Pull request merged**